### PR TITLE
Add country of Origin filter to NomenclaturesController

### PR DESCRIPTION
**Pull request by alexdesi**
### Jira link
https://transformuk.atlassian.net/browse/HOTT-4838

### What?
Add origin filter to `GoodsNomenclaturesController
This PR also contains small refactors to clean up the code/specs.

### Why?
It's a requirement. We need to be able to filter by country of origin.

### CODE REVIEW

File: app/controllers/api/v2/green_lanes/goods_nomenclatures_controller.rb [hunk 1]
```diff
@@ -4,12 +4,20 @@ module GreenLanes
...
-          serializer = Api::V2::GreenLanes::GoodsNomenclatureSerializer.new(presented_gn, include: %w[applicable_measures possible_categorisations])
+          serializer = Api::V2::GreenLanes::GoodsNomenclatureSerializer.new(
+            presented_gn, include: %w[applicable_measures possible_categorisations]
+          )
 
           render json: serializer.serializable_hash
         end
+
+        def param_origin
+          params.fetch(:origin, nil)
```
Comment (jebw) [inline app/controllers/api/v2/green_lanes/goods_nomenclatures_controller.rb:19]: This doesn't handle a blank param, ie `?origin=`  - can this append `.presence`

**Review by jebw — CHANGES_REQUESTED** [review]
Code looks great - minor comment on the param

_but_ I forgot to update the ticket when I changed the API spec so can the param name change from

`origin` to `filters[geographical_area_id]` - this brings us inline with other parts of the API

my bad :man_facepalming:

File: spec/requests/api/v2/green_lanes/goods_nomenclatures_controller_spec.rb [hunk 2]
```diff
@@ -82,13 +87,31 @@
...
     end
+
+    context 'when origin is provided' do
+      let(:params) { { origin: 'AU' } }
+
+      before do
+        allow(GreenLanes::FindCategorisationsService).to receive(:call).and_call_original
+      end
+
+      it 'call FindCategorisationsService with correct params' do
```
Comment (alexdesi) [inline spec/requests/api/v2/green_lanes/goods_nomenclatures_controller_spec.rb:98]: ops:
```
it 'calls ...`
...
```

**Review by rasikasri — APPROVED** [review]

File: app/controllers/api/v2/green_lanes/goods_nomenclatures_controller.rb (hunk 1 above)
Comment (alexdesi) [reply app/controllers/api/v2/green_lanes/goods_nomenclatures_controller.rb]: this is not outdated,  we are using `filter` now.

**Pull request merged**